pkg/dependencies/factory.go

```diff
@@ -159,14 +159,14 @@ func (f *Factory) WithProvider(clusterConfigFile string, clusterConfig *v1alpha1
 	return f
 }
 
-func (f *Factory) WithProviderFactory(clusterConfig *v1alpha1.Cluster) *Factory {
+func (f *Factory) WithProviderFactory(clusterConfig *v1alpha1.Cluster, clusterConfigFile string) *Factory {
```

Comment:
nit: we could simplify this and just pass the clusterConfigFile, since that one already contains the cluster definition